lib/src/sv_sign.c

```diff
@@ -808,9 +804,8 @@ initialize_onvif(signed_video_t *self)
   }
   SV_DONE(status);
   // Free allocated memory
-  free(private_key);
-
-  return status;
+  self->private_key = NULL;
+  if (status != SV_OK && self->onvif != NULL) assert(true);
```

Comment:
Well, you should also check the opposite. I only gave you an example. And also shouldn't it be false in assert?
```
assert((status == SV_OK && !self->onvif) || (status != SV_OK && self->onvif));
```